File: csharp-crud/DTOs/CreateProductDto.cs
using System.ComponentModel.DataAnnotations;

namespace CsharpCrud.DTOs;

public class CreateProductDto
{
    [Required(ErrorMessage = "Nome é obrigatório")]
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    [Required]
    [Range(0.01, double.MaxValue, ErrorMessage = "Preço deve ser maior que zero")]
    public decimal Price { get; set; }

    public int? Quantity { get; set; }
} 

File: typescript-crud/src/products/dto/create-product.dto.ts
import { IsNotEmpty, IsNumber, IsOptional, IsPositive, IsString } from 'class-validator';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class CreateProductDto {
  @ApiProperty({ description: 'Nome do produto' })
  @IsNotEmpty({ message: 'Nome é obrigatório' })
  @IsString()
  name: string;

  @ApiPropertyOptional({ description: 'Descrição do produto' })
  @IsOptional()
  @IsString()
  description?: string;

  @ApiProperty({ description: 'Preço do produto' })
  @IsPositive({ message: 'Preço deve ser maior que zero' })
  @IsNumber()
  price: number;

  @ApiPropertyOptional({ description: 'Quantidade em estoque' })
  @IsOptional()
  @IsNumber()
  quantity?: number;
} 